cupyx/scipy/sparse/linalg/_solve.py

```diff
@@ -78,6 +80,300 @@ def lsqr(A, b):
     return ret
 
 
+def lsmr(A, b, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8, maxiter=None):
+    """Iterative solver for least-squares problems.
+
+    lsmr solves the system of linear equations ``Ax = b``. If the system
+    is inconsistent, it solves the least-squares problem ``min ||b - Ax||_2``.
+    A is a rectangular matrix of dimension m-by-n, where all cases are
+    allowed: m = n, m > n, or m < n. B is a vector of length m.
+    The matrix A may be dense or sparse (usually sparse).
+
+    Args:
+        A (ndarray, spmatrix or LinearOperator): The real or complex
+            matrix of the linear system. ``A`` must be
+            :class:`cupy.ndarray`, :class:`cupyx.scipy.sparse.spmatrix` or
+            :class:`cupyx.scipy.sparse.linalg.LinearOperator`.
+        b (cupy.ndarray): Right hand side of the linear system with shape
+            ``(m,)`` or ``(m, 1)``.
+        damp (float): Damping factor for regularized least-squares.
+            `lsmr` solves the regularized least-squares problem
+            ::
+
+                min ||(b) - (  A   )x||
+                    ||(0)   (damp*I) ||_2
+
+            where damp is a scalar. If damp is None or 0, the system
+            is solved without regularization.
+        atol, btol (float):
+            Stopping tolerances. `lsmr` continues iterations until a
+            certain backward error estimate is smaller than some quantity
+            depending on atol and btol.
+        conlim (float): `lsmr` terminates if an estimate of ``cond(A)`` i.e.
+            condition number of matrix exceeds `conlim`. If `conlim` is None,
+            the default value is 1e+8.
+        maxiter (int): Maximum number of iterations.
+
+    Returns:
+        tuple:
+            - `x` (ndarray): Least-square solution returned.
+            - `istop` (int): istop gives the reason for stopping
```

Comment:
Would you make the detail of `istop` description a literal block?
```suggestion
            - `istop` (int): istop gives the reason for stopping::
```